File: leetcode/data_struct/array/0035/1.ts
// auther: phoenix
// date: 2023-06-12 09:40:17

/*


*/

demo()
function demo() {
  searchInsert1Test()
}


function searchInsert1Test(){
  console.log(searchInsert1([1,3,5,6], 5)) // 2
  console.log(searchInsert1([1,3,5,6], 2)) // 1
  console.log(searchInsert1([1,3,5,6], 7)) // 4
  console.log(searchInsert1([1,3,5,6], 0)) // 0
}
function searchInsert1(nums: number[], target: number):number{
  let index:number = 0
  const length:number = nums.length
  const contain = nums.includes(target)
  for (let i = 0; i < length; i++) {
    const el = nums[i]
    if (contain && target === el) {
      return i
    }
    if (!contain && target >= el) {
      index =  i + 1
    }
  }
  return index
}

File: leetcode/data_struct/array/0035/1.rs
// auther: phoenix
// date: 2023-06-12 09:40:14

/*



*/

fn main() {
  search_insert1_test();
}

fn search_insert1_test() {
  println!("{}", search_insert1(vec![1,3,5,6], 5)); // 2
  println!("{}", search_insert1(vec![1,3,5,6], 2)); // 1
  println!("{}", search_insert1(vec![1,3,5,6], 7)); // 4
  println!("{}", search_insert1(vec![1,3,5,6], 0)); // 0
}

fn search_insert1(nums: Vec<isize>, target: isize) -> usize {
  let mut index:usize = 0;
  let length:usize = nums.len();
  let contain: bool = nums.contains(&target);
  for i in 0..length {
    if contain && nums.get(i).unwrap() == &target {
      return i;
    }
    if !contain && nums.get(i).unwrap() <= &target {
      index = i + 1;
    }
  }
  return index;
}